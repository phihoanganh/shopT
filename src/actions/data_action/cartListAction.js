import CartListRepository from '../../network/respository/cartListRepository';
import { CartListActionType } from '../../action_type'
import Const from '../../global/const'
import Utils from '../../global/utils'

export function getCartListSuccess(data) {
    return {
        type: CartListActionType.GET_CART_LIST_SUCCESS,
        data
    }
}

export function getCartListFail() {
    return { type: CartListActionType.GET_CART_LIST_FAIL }
}

export function buyConfirm() {
    return { type: CartListActionType.BUY_CONFIRM }
}

export function doGetCartList() {
    return dispatch => {
        CartListRepository.getCartList().then(data => {
            dispatch(getCartListSuccess(data))
        })
    };
}

export function doAddCartItem(item, cartList) {
    return dispatch => {
        let itemIndex = cartList.findIndex((element) => {
            return element.id === item.id
        })
        if (itemIndex != -1) {
            if (cartList[itemIndex].num >= cartList[itemIndex].stock) {
                cartList[itemIndex].num = cartList[itemIndex].stock
            } else {
                cartList[itemIndex].num++
            }
        } else {
            let temp = item
            temp.num = 1
            cartList.push(temp)
        }
        dispatch(getCartListSuccess(cartList))
        Utils.setValueByKey(Const.STORE_ID_CART_LIST, JSON.stringify(cartList))
    }
}

export function doMinusCartItem(item, cartList) {
    return dispatch => {
        let itemIndex = cartList.findIndex((element) => {
            return element.id === item.id
        })
        if (cartList[itemIndex].num <= 1) {
            cartList.splice(itemIndex, 1)
        } else {
            cartList[itemIndex].num--
        }

        dispatch(getCartListSuccess(cartList))
        Utils.setValueByKey(Const.STORE_ID_CART_LIST, JSON.stringify(cartList))
    }
}
export function doRemoveCartItem(index, cartList) {
    return dispatch => {
        cartList.splice(index, 1)
        dispatch(getCartListSuccess(cartList))
        Utils.setValueByKey(Const.STORE_ID_CART_LIST, JSON.stringify(cartList))
    }
}

export function clearCartItem() {
    return dispatch => {
        dispatch(buyConfirm())
        Utils.setValueByKey(Const.STORE_ID_CART_LIST, '')
    }
}