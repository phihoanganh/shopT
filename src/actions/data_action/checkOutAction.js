import CheckOutRepository from '../../network/respository/checkOutRepository';
import { CheckOutActionType } from '../../action_type'
import Const from '../../global/const'
import Utils from '../../global/utils'

export function getCheckOutInfoSuccess(data) {
    return {
        type: CheckOutActionType.GET_CHECK_OUT_INFO_SUCCESS,
        data
    }
}

export function getCheckOutInfoFail() {
    return { type: CheckOutActionType.GET_CHECK_OUT_INFO_FAIL }
}

export function doGetCheckOutInfo() {
    return dispatch => {
        return CheckOutRepository.getCheckOutInfo().then(data => {
            dispatch(getCheckOutInfoSuccess(data))
        })
    };
}
export function doSetCheckOutInfo(data) {
    return dispatch => {
        dispatch(getCheckOutInfoSuccess(data))
        Utils.setValueByKey(Const.STORE_ID_CHECK_OUT, JSON.stringify(data))
    };
}