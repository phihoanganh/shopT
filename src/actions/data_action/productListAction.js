import ProductListRepository from '../../network/respository/productListRepository';
import { ProductListActionType } from '../../action_type'
import Const from '../../global/const'
import Utils from '../../global/utils'

export function getProductListSuccess(data) {
    return {
        type: ProductListActionType.GET_PRODUCT_LIST_SUCCESS,
        data
    }
}

export function getProductListFail() {
    return { type: ProductListActionType.GET_PRODUCT_LIST_FAIL }
}

export function doGetProductList() {
    return dispatch => {
        return ProductListRepository.getProductList().then(data => {
            dispatch(getProductListSuccess(data))
        })
    };
}

export function doBuyProduct(buyList, productList) {
    return dispatch => {
        buyList.forEach(buyElement => {
            let itemIndex = productList.findIndex((element) => {
                return buyElement.id === element.id
            })
            if (itemIndex != -1) {
                productList[itemIndex].stock -= buyElement.num
            }
        })
        dispatch(getProductListSuccess(productList))
        Utils.setValueByKey(Const.STORE_ID_PRODUCT_LIST, JSON.stringify(productList))
    }
}