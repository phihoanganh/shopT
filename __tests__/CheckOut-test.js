import React from 'react';
import { shallow } from 'enzyme';
import Checkout from '../src/views/checkout';
import Enzyme from 'enzyme';
import thunk from 'redux-thunk'
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import * as actions from '../src/actions/data_action/checkOutAction'
import types from '../src/action_type/data_type/checkOutActionType'

Enzyme.configure({ adapter: new Adapter() });
const middlewares = [thunk]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);
jest.mock('react-navigation', () => ({ withNavigation: component => component }));

const initialState = {

};

describe('CheckOut Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Checkout />, { context: { store: mockStore(initialState) } },
      )
      expect(component).toMatchSnapshot()
    })
  })
  describe('actions', () => {
    let data = {
      ship_adss: { name: 'AAA', address: '222 AAA', phone: '093837822', note: '' },
      pay_method: { isVisa: false }
    }
    const store = mockStore({});
    it('doGetCheckOutInfo Success', () => {
      store.dispatch(actions.doSetCheckOutInfo(data));
      expect(store.getActions()).toContainEqual({
        type: types.GET_CHECK_OUT_INFO_SUCCESS,
        data
      });
    })
    it('doGetCheckOutInfo Success', async () => {
      await store.dispatch(actions.doGetCheckOutInfo());
      expect(store.getActions()).toContainEqual({
        type: types.GET_CHECK_OUT_INFO_SUCCESS,
        data
      });
    })
  })
})