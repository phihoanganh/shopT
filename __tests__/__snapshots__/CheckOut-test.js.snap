// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOut Component Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
